<launch>

    <arg name="namespace" default="sam"/>

    <!--
    <node name="pid_setpoint_static" pkg="sam_loop_test" type="pid_setpoint_static" ns="LCG_trim">
      <param name="setpoint_topic" value="/pitch_setpoint" />
      <param name="setpoint" value="0.15" />
    </node>

    <node name="pid_setpoint_static" pkg="sam_loop_test" type="pid_setpoint_static" ns="VBS_depth">
      <param name="setpoint_topic" value="/depth_setpoint" />
      <param name="setpoint" value="50" />
    </node>
    -->

    <node name="pid_tf_listener" pkg="sam_loop_test" type="pid_tf_listener" output="screen" >
	    <param name="odom_frame" value="$(arg namespace)_odom"/>
	    <param name="base_frame" value="sam_auv_1/base_link"/>
	    <param name="world_frame" value="world"/>
    </node>

    <node name="pid_gazebo" pkg="sam_loop_test" type="pid_gazebo" ns="LCG_trim" >
      <param name="topic_from_controller" value="lcg_control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="limit_between_setpoints" value="0.01" />
      <remap from="/actuator_input" to="/sam_auv_1/joint2_position_controller/command"/>
    </node>

    <node name="pid_gazebo" pkg="sam_loop_test" type="pid_gazebo" ns="VBS_depth" >
      <param name="topic_from_controller" value="vbs_control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="limit_between_setpoints" value="0.01" />
      <remap from="/actuator_input" to="/sam_auv_1/joint1_position_controller/command"/>
    </node>


    <!--node name="pid_gazebo" pkg="sam_loop_test" type="pid_gazebo" ns="TCG_roll" >
      <param name="topic_from_controller" value="vbs_control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="limit_between_setpoints" value="1" />
      <remap from="/actuator_input" to="/sam_auv_1/joint1_position_controller/command"/>
    </node-->

    <node name="controller" pkg="pid" type="controller" ns="LCG_trim" output="screen" >
      <param name="Kp" value="5" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.1" />
      <param name="upper_limit" value="1.5" />
      <param name="lower_limit" value="-1.5" />
      <param name="windup_limit" value="10" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
      <param name="topic_from_plant" value="/pitch_feedback" />
      <param name="topic_from_controller" value="lcg_control_action" />
      <remap from="setpoint" to="/pitch_setpoint" />
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
     </node>

    <!--node name="controller" pkg="pid" type="controller" ns="VBS_depth" output="screen" >
      <param name="Kp" value="5.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.1" />
      <param name="upper_limit" value="1.5" />
      <param name="lower_limit" value="-1.5" />
      <param name="windup_limit" value="100" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" /-->
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
      <!--param name="topic_from_plant" value="/depth_feedback" />
      <param name="topic_from_controller" value="vbs_control_action" />
      <remap from="setpoint" to="/depth_setpoint" /-->
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
     <!--/node-->

</launch>
